### PR TITLE
Huobi

### PR DESCRIPTION
**Pull request by belerweb**

### CODE REVIEW

File: service-netty/pom.xml [hunk 1]
```diff
@@ -28,5 +28,10 @@
             <artifactId>jackson-databind</artifactId>
             <version>2.9.5</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>1.3.2</version>
+        </dependency>
```
Comment (Flemingjp) [inline service-netty/pom.xml]: Is it really required to add this dependency. I'm not against adding new dependencies, but once it is added it is very difficult to remove in the future.

Comment (Flemingjp) [inline service-netty/pom.xml]: Is it really required to add this dependency? I'm not against adding new dependencies, but once it is added it is very difficult to remove in the future.

File: xchange-huobi/pom.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        <dependency>
+            <groupId>org.knowm.xchange</groupId>
+            <artifactId>xchange-huobi</artifactId>
+            <version>${xchange.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.alibaba</groupId>
+            <artifactId>fastjson</artifactId>
+            <version>1.2.47</version>
+        </dependency>
```
Comment (Flemingjp) [inline xchange-huobi/pom.xml]: Already the Jackson JSON library as a dependency, isn't this redundant?

**Comment by amit2103** [comment]
@Flemingjp  Is it possible to merge this now ? Huobi will be a good addition

**Comment by Flemingjp** [comment]
@amit2103 are you planning on using this exchange, if so would you be able to test this PR and if possible maintain it in the future?

**Comment by amit2103** [comment]
I can definitely help to test no issues...I am not sure about maintaining honestly

**Comment by Flemingjp** [comment]
@amit2103  If you could test and then submit a review, Ill happily merge this PR

**Comment by amit2103** [comment]
I have tried to run teh original code. It doesnt seem to work , and I keep on getting this traces 

13:14:00.741 [nioEventLoopGroup-92-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.detail","id":1543304640741}
13:14:00.741 [nioEventLoopGroup-92-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.trade.detail","id":1543304640741}
13:14:26.722 [nioEventLoopGroup-92-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client received closing
13:14:26.723 [nioEventLoopGroup-92-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client disconnected!
13:14:26.723 [nioEventLoopGroup-92-1] INFO  info.bitrich.xchangestream.huobi.HuobiStreamingService - Reopening websocket because it was closed by the host
13:14:28.941 [globalEventExecutor-1-95] INFO  info.bitrich.xchangestream.huobi.HuobiStreamingService - Connecting to wss://api.huobi.pro:-1/ws
13:14:29.617 [nioEventLoopGroup-93-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client connected!
13:14:29.617 [nioEventLoopGroup-93-1] WARN  info.bitrich.xchangestream.huobi.HuobiStreamingService - Resubscribing channels
13:14:29.617 [nioEventLoopGroup-93-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.depth.step0","id":1543304669617}
13:14:29.617 [nioEventLoopGroup-93-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.detail","id":1543304669617}
13:14:29.618 [nioEventLoopGroup-93-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.trade.detail","id":1543304669618}
13:14:47.785 [nioEventLoopGroup-93-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client received closing
13:14:47.785 [nioEventLoopGroup-93-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client disconnected!
13:14:47.785 [nioEventLoopGroup-93-1] INFO  info.bitrich.xchangestream.huobi.HuobiStreamingService - Reopening websocket because it was closed by the host
13:14:50.033 [globalEventExecutor-1-96] INFO  info.bitrich.xchangestream.huobi.HuobiStreamingService - Connecting to wss://api.huobi.pro:-1/ws
13:14:51.136 [nioEventLoopGroup-94-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client connected!
13:14:51.136 [nioEventLoopGroup-94-1] WARN  info.bitrich.xchangestream.huobi.HuobiStreamingService - Resubscribing channels
13:14:51.136 [nioEventLoopGroup-94-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.depth.step0","id":1543304691136}
13:14:51.137 [nioEventLoopGroup-94-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.detail","id":1543304691137}
13:14:51.137 [nioEventLoopGroup-94-1] DEBUG info.bitrich.xchangestream.huobi.HuobiStreamingService - Sending message: {"sub":"market.ethbtc.trade.detail","id":1543304691137}

**Comment by amit2103** [comment]
It seems that huobi keeps closing the websocket and we never receive any messages. @belerweb  take a look

**Comment by badgerwithagun** [comment]
@belerweb, are you able to follow up on this?

**Comment by badgerwithagun** [comment]
Closing as stale. @belerweb or @mdvx, it'd be great if you could re-open the PR with @mdvx's proposed fixes and we can get this merged.

**Comment by mdvx** [comment]
I am looping back to Huobi Global https://huobiapi.github.io/docs/spot/v1/en/#introduction
I see we once had xchange-huobi, does anybody know where this is now located?

@badgerwithagun @Flemingjp @belerweb @amit2103

**Comment by Flemingjp** [comment]
@mdvx From what I can tell xchange-huobi was never merged into xchange-streams, as this pull request was never completed.

[XChange](https://github.com/knowm/XChange/tree/develop/xchange-huobi) has a REST API implementation.

**Comment by mdvx** [comment]
Whats the best way to resurrect this? (I could apply https://patch-diff.githubusercontent.com/raw/bitrich-info/xchange-stream/pull/249.patch to my branch, but that would lose @Flemingjp contrib record.  @badgerwithagun @TSavo

**Comment by Flemingjp** [comment]
@mdvx That is one possible option. Or we could reopen this pull request if you are able/willing to clean up the current progress. @badgerwithagun What are your thoughts?

**Comment by mdvx** [comment]
I can clean up, and I want to add https://huobiapi.github.io/docs/spot/v1/en/#websocket-asset-and-order-v2 (worried about the signature thing, but will figure it out)

**Comment by badgerwithagun** [comment]
I really don't mind as long as there's a clean diff at the end of it to review :)

Feel free to re-open and rebase this one, or clone it and resubmit.

**Comment by badgerwithagun** [comment]
OK, I'm going to close this to avoid splitting the conversation.

**Pull request closed without merging**